src/test/java/uk/gov/hmcts/reform/managecase/repository/DataStoreRepositoryTest.java

```diff
@@ -226,4 +262,151 @@ void shouldFindCaseByCaseId() {
+
+        // ASSERT
+        verify(dataStoreApi).getStartEventTrigger(CASE_ID, EVENT_ID);
+        ArgumentCaptor<CaseDataContent> captor = ArgumentCaptor.forClass(CaseDataContent.class);
+        verify(dataStoreApi).submitEventForCase(any(String.class), captor.capture());
+
+        CaseDataContent caseDataContentCaptorValue = captor.getValue();
+        assertThat(caseDataContentCaptorValue.getToken()).isEqualTo(EVENT_TOKEN);
+
+        assertThat(caseDataContentCaptorValue.getEvent().getDescription()).isEqualTo(NOC_REQUEST_DESCRIPTION);
+        assertThat(caseDataContentCaptorValue.getEvent().getEventId()).isEqualTo(EVENT_ID);
+
+        assertThat(caseDataContentCaptorValue.getData().get(CHANGE_ORGANISATION_REQUEST_FIELD).toString())
+            .isEqualTo(EXPECTED_NOC_REQUEST_DATA);
+    }
+
+    @Test
+    @DisplayName("find case by id using external facing API")
+    void shouldFindCaseByIdUsingExternalApi() {
+        // ARRANGE
+        CaseResource caseResource = CaseResource.builder().build();
+        given(dataStoreApi.getCaseDetailsByCaseIdViaExternalApi(CASE_ID)).willReturn(caseResource);
+
+        // ACT
+        CaseResource result = repository.findCaseByCaseIdExternalApi(CASE_ID);
+
+        // ASSERT
+        assertThat(result).isEqualTo(caseResource);
+        verify(dataStoreApi).getCaseDetailsByCaseIdViaExternalApi(eq(CASE_ID));
+    }
+
+    @Test
+    @DisplayName("handle missing event token when calling submit event for case")
+    void shouldThrowExceptionWhenSubmitEventForCaseCalledWithoutEventToken() {
+
+        // ARRANGE
+        CaseUpdateViewEvent caseUpdateViewEvent = CaseUpdateViewEvent.builder()
+            .caseFields(getCaseViewFields())
+            .wizardPages(getWizardPages("testCVaseField"))
+            .build();
+
+        given(dataStoreApi.getStartEventTrigger(CASE_ID, EVENT_ID)).willReturn(caseUpdateViewEvent);
+
+        // ACT & ASSERT
+        IllegalStateException illegalStateException = assertThrows(IllegalStateException.class, () ->
+            repository.submitEventForCase(CASE_ID, EVENT_ID, ChangeOrganisationRequest.builder().build())
+        );
+
+        assertThat(illegalStateException.getMessage())
+            .isEqualTo(NOT_ENOUGH_DATA_TO_SUBMIT_START_EVENT);
+
+        verify(dataStoreApi, never()).submitEventForCase(any(), any());
+    }
+
+    @Test
+    @DisplayName("handle missing approval status default value when calling submit event for case")
+    void shouldSetDefaultApprovalStatusWhenSubmitEventForCaseCalledWithoutDefaultValue() {
+
+        // ARRANGE
+        CaseUpdateViewEvent caseUpdateViewEvent = CaseUpdateViewEvent.builder()
+            .caseFields(getCaseViewFields())
+            .eventToken(EVENT_TOKEN)
+            .wizardPages(Collections.emptyList())
+            .build();
+
+        given(dataStoreApi.getStartEventTrigger(CASE_ID, EVENT_ID)).willReturn(caseUpdateViewEvent);
+
+
+    }
```

Comment:
Is this test complete: i.e. where is the act and assert, or am I not reading it correctly?

Reply:
Was a WIP checked in by mistake, will remove